btree.go

```diff
@@ -157,6 +157,13 @@ func (btree *btreeTable) decodeRecord(payload []byte) (Record, error) {
 			recbuf = recbuf[int(n):]
 			v = vv
 
+		case StInt24:
+			bs := make([]byte, 3)
+			n64, n := varint(bs)
```

Comment:
These types are not varints. Varints are variable-length (signaled by high bit in each byte), whereas these are fixed width. I would probably either write my own big-endian unmarshal routine that operates on `[]byte` slices (multiply sum by 256, add next byte, repeat), or copy into the next largest slice that fits the `binary.ByteOrder` unmarshaling routines sizes (32,64).

Reply:
I figured that was too easy. Would this be better?

```go
	case StInt24:
		n := 3 // bytes
		bs := make([]byte, n)
		copy(bs, recbuf[:n])
		recbuf = recbuf[n:]
		r := bytes.NewReader(bs)
		var vv int64
		err := binary.Read(r, binary.BigEndian, vv)
		if err != nil {
			panic(err)
		}
		v = vv
```

Reply:
I would just do something simpler like:

```go
bs := make([]byte, 4)
if n := copy(bs[1:], recbuf); n != 3 {
  // return an error
}
vv = binary.BigEndian.Uint32(bs)
```

Reply:
agreed. modulo, perhaps, `binary.BigEndian.Int32(bs)` (I think `StInt24` are signed, right?)

Reply:
Hmmmm. Probably. If so, you're going to have to sign-extend bit 24 into bits 24-32. So if bs[1]&0x80 > 0, set bs[0] to 0xff.

Reply:
Either way, a couple of targeted tests should convince you that you've done it correctly :-)

Reply:
These are from Section 2.1, right? https://sqlite.org/fileformat.html#record_format

If so, they're twos-complement, aka signed ints.

Reply:
Yeah, sqlite seems to use signed ints everywhere.